platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/signing/DefaultSelfEventSigner.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.signing;
+
+import com.swirlds.common.crypto.Signature;
+import com.swirlds.platform.crypto.KeysAndCerts;
+import com.swirlds.platform.crypto.PlatformSigner;
+import com.swirlds.platform.event.GossipEvent;
+import com.swirlds.platform.system.events.BaseEventHashedData;
+import com.swirlds.platform.system.events.BaseEventUnhashedData;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+
+/**
+ * A default implementation of {@link SelfEventSigner}.
+ */
+public class DefaultSelfEventSigner implements SelfEventSigner {
+
+    private final KeysAndCerts keysAndCerts;
+
+    /**
+     * Constructor.
+     *
+     * @param keysAndCerts the platform's keys and certificates
+     */
+    public DefaultSelfEventSigner(@NonNull final KeysAndCerts keysAndCerts) {
+        this.keysAndCerts = Objects.requireNonNull(keysAndCerts);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public GossipEvent signEvent(@NonNull final BaseEventHashedData event) {
+        final Signature signature = new PlatformSigner(keysAndCerts).sign(event.getHash());
+        final BaseEventUnhashedData unhashedData = new BaseEventUnhashedData(null, signature.getSignatureBytes());
```

Comment:
Why not populate `otherId` with the correct value?

Reply:
![unhashedData](https://github.com/hashgraph/hedera-services/assets/56973212/2e7a806c-7a18-44bd-9eae-b99f1f987599)

Reply:
We never should have written the other parent ID to this location in the first place. We can't remove the field in the short term due to the overhead of data migration, but that doesn't mean we need to write anything to it. Anything that still reads this field should be considered now and going forward.

Reply:
Gotcha